pkg/apis/core/validation/validation.go

```diff
@@ -5284,6 +5359,11 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 	return allErrs
 }
 
+func isPodLevelResourcesSet(pod *core.Pod) bool {
```

Comment:
Really really minor OK to disregard, but I had to go search for callers to make sure this didn't need to handle unsupported keys.  

This could be a func literal at the callsite, to emphasize that it is HYPER local.  

```
    isPodLevelResourcesSet := func(pod *core.Pod) bool {
        return pod.Spec.Resources != nil &&
            len(pod.Spec.Resources.Requests) + len(pod.Spec.Resources.Limits) > 0
    }
    if isPodLevelResourcesSet(oldPod) || isPodLevelResourcesSet(newPod) {
        return...
```